ksqldb-engine/src/main/java/io/confluent/ksql/function/udaf/offset/LatestByOffset.java

```diff
@@ -50,6 +52,7 @@ private LatestByOffset() {
   }
 
   static AtomicLong sequence = new AtomicLong();
+  private static Comparator<Struct> comparator;
```

Comment:
we should not be relying on static state - if people create two different latest by offests (one to ignore nulls and one not to) whichever is created second will override the comparator for the first. Instead, it should be created when `latestN` method is called as a local variable and referenced in the anonymous class 

In general, if you see code with static, non-final state it's probably a bug and best to avoid.